### PR TITLE
fix: #1735 Unable to use models with a StreamField inside Inline Panels

### PR DESCRIPTION
**Pull request by Gagaro**
Please see #1735 for more discussion about this.

### CODE REVIEW

**Comment by Gagaro** [comment]
Any idea why the build fail?

```
$ git fetch --no-tags --depth=50 origin +refs/pull/2659/merge:
fatal: Couldn't find remote ref refs/pull/2659/merge
[info] build failed (exit code 1)
```

**Comment by kaedroho** [comment]
Looks like it was trying to get a "merged" copy of the PR which doesn't exist as this PR isn't mergable.

rebasing or merging master (resolving the conflict) should solve the error

**Comment by Gagaro** [comment]
Indeed, there was a commit one minute after I created the PR :sweat_smile: . Thanks :+1: .

**Comment by Gagaro** [comment]
There is a regression with this PR. Some media are missing when they are needed in the FieldPanel but not in the InlinePanel.

```
Uncaught ReferenceError: makeHalloRichTextEditable is not defined
```

**Comment by Gagaro** [comment]
Fixed

**Comment by m1kola** [comment]
Hi @Gagaro!

Issue with `InlinePanel` was fixed in #2673. Thanks!

**Comment by Gagaro** [comment]
Nice, I updated the pull request to only include the things specific to StreamField.

File: wagtail/wagtailadmin/edit_handlers.py [hunk 1]
```diff
@@ -621,6 +621,16 @@ def required_formsets(cls):
...
         }
 
+    @classmethod
+    def html_declarations(cls):
+        # We need to include html_declarations for StreamFields in the related model
+        declarations = [
+            field.stream_block.all_html_declarations()
+            for field in cls.related.related_model._meta.fields
+            if hasattr(field, 'stream_block')
+        ]
```
Comment (gasman) [inline wagtail/wagtailadmin/edit_handlers.py]: We really shouldn't have any special-case code for StreamField (or any other field type) here - that would imply that something in the StreamField mechanism is failing to follow the same spec as everything else. Investigating...

Comment (gasman) [inline wagtail/wagtailadmin/edit_handlers.py]: Looks like all we really need here is `return cls.get_child_edit_handler_class().html_declarations()`. Packaging this up with some extra tests now...

**Comment by gasman** [comment]
Merged in 16c279d705dfd944078a1c6e0e13c610d547e5e2. Thanks!

**Pull request closed without merging**